File: Smallest Positive missing number[9-July-2023]/solution.py
#  >> PYTHON CODE <<

class Solution:
    
    #Function to find the smallest positive number missing from the array.
    def missingNumber(self,arr,n):
        
        i=0
        
        while(i<n):
            el=arr[i]
            index=el-1
            
            if(index>=0 and index<n):
                if(arr[index]!=el):
                    arr[index],arr[i]=arr[i],arr[index]
                    continue
            i+=1
        
        for i in range(n):
            if(arr[i]!=i+1):
                return i+1
        
        return n+1

File: Smallest Positive missing number[9-July-2023]/solution.cpp
//  >> C++ CODE <<

class Solution
{
    public:
    //Function to find the smallest positive number missing from the array.
    int missingNumber(int arr[], int n) 
    { 
        int i = 0;

        while (i < n) {
            int el = arr[i];
            int index = el - 1;

            if (index >= 0 && index < n) {
                if (arr[index] != el) {
                    int temp = arr[index];
                    arr[index] = arr[i];
                    arr[i] = temp;
                    continue;
                }
            }
            i++;
        }

        for (i = 0; i < n; i++) {
            if (arr[i] != i + 1) {
                return i + 1;
            }
        }

        return n + 1;
    } 
};